msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/AbstractApplicationBase.java

```diff
@@ -0,0 +1,476 @@
+
+        AccountsSupplier supplier = new AccountsSupplier(this, msalRequest);
+
+        return this.serviceBundle.getExecutorService() != null ?
+                CompletableFuture.supplyAsync(supplier, serviceBundle.getExecutorService()) :
+                CompletableFuture.supplyAsync(supplier);
+    }
+
+    public CompletableFuture<Void> removeAccount(IAccount account) {
+        RequestContext context = new RequestContext(this, PublicApi.REMOVE_ACCOUNTS, null);
+        MsalRequest msalRequest = new MsalRequest(this, null, context) {
+        };
+
+        RemoveAccountRunnable runnable = new RemoveAccountRunnable(msalRequest, account);
+
+        return serviceBundle.getExecutorService() != null ?
+                CompletableFuture.runAsync(runnable, serviceBundle.getExecutorService()) :
+                CompletableFuture.runAsync(runnable);
+    }
+
+    CompletableFuture<IAuthenticationResult> executeRequest(
+            MsalRequest msalRequest) {
+
+        AuthenticationResultSupplier supplier = getAuthenticationResultSupplier(msalRequest);
+
+        ExecutorService executorService = serviceBundle.getExecutorService();
+        return executorService != null ?
+                CompletableFuture.supplyAsync(supplier, executorService) :
+                CompletableFuture.supplyAsync(supplier);
+
+    }
+
+    AuthenticationResult acquireTokenCommon(MsalRequest msalRequest, Authority requestAuthority)
+            throws Exception {
+
+        HttpHeaders headers = msalRequest.headers();
+
+        if (logPii) {
+            log.debug(LogHelper.createMessage(
+                    String.format("Using Client Http Headers: %s", headers),
+                    headers.getHeaderCorrelationIdValue()));
+        }
+
+        TokenRequestExecutor requestExecutor = new TokenRequestExecutor(
+                requestAuthority,
+                msalRequest,
+                serviceBundle);
+
+        AuthenticationResult result = requestExecutor.executeTokenRequest();
+
+        if (authenticationAuthority.authorityType.equals(AuthorityType.AAD)) {
+            InstanceDiscoveryMetadataEntry instanceDiscoveryMetadata =
+                    AadInstanceDiscoveryProvider.getMetadataEntry(
+                            requestAuthority.canonicalAuthorityUrl(),
+                            validateAuthority,
+                            msalRequest,
+                            serviceBundle);
+
+            tokenCache.saveTokens(requestExecutor, result, instanceDiscoveryMetadata.preferredCache);
+        } else {
+            tokenCache.saveTokens(requestExecutor, result, authenticationAuthority.host);
+        }
+
+        return result;
+    }
+
+    @Override
+    public CompletableFuture<IAuthenticationResult> acquireToken(RefreshTokenParameters parameters) {
+
+        validateNotNull("parameters", parameters);
+
+        RequestContext context = new RequestContext(
+                this,
+                PublicApi.ACQUIRE_TOKEN_BY_REFRESH_TOKEN,
```

Comment:
I don't think this is part of the base. It's maybe part of the "client application". The base should not have any AcquireToken* methods.